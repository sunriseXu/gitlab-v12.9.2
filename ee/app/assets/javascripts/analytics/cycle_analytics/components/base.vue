<script>
import { GlEmptyState, GlLoadingIcon } from '@gitlab/ui';
import { mapActions, mapState, mapGetters } from 'vuex';
import { featureAccessLevel } from '~/pages/projects/shared/permissions/constants';
import glFeatureFlagsMixin from '~/vue_shared/mixins/gl_feature_flags_mixin';
import { PROJECTS_PER_PAGE } from '../constants';
import GroupsDropdownFilter from '../../shared/components/groups_dropdown_filter.vue';
import ProjectsDropdownFilter from '../../shared/components/projects_dropdown_filter.vue';
import Scatterplot from '../../shared/components/scatterplot.vue';
import { LAST_ACTIVITY_AT, dateFormats, DATE_RANGE_LIMIT } from '../../shared/constants';
import DateRange from '../../shared/components/daterange.vue';
import StageDropdownFilter from './stage_dropdown_filter.vue';
import SummaryTable from './summary_table.vue';
import StageTable from './stage_table.vue';
import TasksByTypeChart from './tasks_by_type_chart.vue';
import UrlSyncMixin from '../mixins/url_sync_mixin';

export default {
  name: 'CycleAnalytics',
  components: {
    DateRange,
    GlLoadingIcon,
    GlEmptyState,
    GroupsDropdownFilter,
    ProjectsDropdownFilter,
    SummaryTable,
    StageTable,
    StageDropdownFilter,
    Scatterplot,
    TasksByTypeChart,
  },
  mixins: [glFeatureFlagsMixin(), UrlSyncMixin],
  props: {
    emptyStateSvgPath: {
      type: String,
      required: true,
    },
    noDataSvgPath: {
      type: String,
      required: true,
    },
    noAccessSvgPath: {
      type: String,
      required: true,
    },
    hideGroupDropDown: {
      type: Boolean,
      required: true,
    },
  },
  computed: {
    ...mapState([
      'featureFlags',
      'isLoading',
      'isLoadingStage',
      'isLoadingTasksByTypeChart',
      'isLoadingDurationChart',
      'isEmptyStage',
      'isSavingCustomStage',
      'isCreatingCustomStage',
      'isEditingCustomStage',
      'selectedGroup',
      'selectedProjects',
      'selectedStage',
      'stages',
      'summary',
      'labels',
      'topRankedLabels',
      'currentStageEvents',
      'customStageFormEvents',
      'errorCode',
      'startDate',
      'endDate',
      'tasksByType',
      'medians',
      'customStageFormErrors',
    ]),
    ...mapGetters([
      'hasNoAccessError',
      'currentGroupPath',
      'durationChartPlottableData',
      'tasksByTypeChartData',
      'durationChartMedianData',
      'activeStages',
      'selectedProjectIds',
    ]),
    shouldRenderEmptyState() {
      return !this.selectedGroup;
    },
    hasCustomizableCycleAnalytics() {
      return Boolean(this.glFeatures.customizableCycleAnalytics);
    },
    shouldDisplayFilters() {
      return this.selectedGroup && !this.errorCode;
    },
    shouldDisplayDurationChart() {
      return this.featureFlags.hasDurationChart && !this.hasNoAccessError;
    },
    shouldDisplayTasksByTypeChart() {
      return this.featureFlags.hasTasksByTypeChart && !this.hasNoAccessError;
    },
    isDurationChartLoaded() {
      return !this.isLoadingDurationChart && !this.isLoading;
    },
    isTasksByTypeChartLoaded() {
      return !this.isLoading && !this.isLoadingTasksByTypeChart;
    },
    hasDateRangeSet() {
      return this.startDate && this.endDate;
    },
    selectedTasksByTypeFilters() {
      const {
        selectedGroup,
        startDate,
        endDate,
        selectedProjectIds,
        tasksByType: { subject, selectedLabelIds },
      } = this;
      return {
        selectedGroup,
        selectedProjectIds,
        startDate,
        endDate,
        subject,
        selectedLabelIds,
      };
    },
  },
  mounted() {
    this.setFeatureFlags({
      hasDurationChart: this.glFeatures.cycleAnalyticsScatterplotEnabled,
      hasDurationChartMedian: this.glFeatures.cycleAnalyticsScatterplotMedianEnabled,
      hasTasksByTypeChart: this.glFeatures.tasksByTypeChart,
    });
  },
  methods: {
    ...mapActions([
      'fetchCycleAnalyticsData',
      'fetchStageData',
      'setSelectedGroup',
      'setSelectedProjects',
      'setSelectedStage',
      'hideCustomStageForm',
      'showCustomStageForm',
      'showEditCustomStageForm',
      'setDateRange',
      'fetchTasksByTypeData',
      'updateSelectedDurationChartStages',
      'createCustomStage',
      'updateStage',
      'removeStage',
      'setFeatureFlags',
      'clearCustomStageFormErrors',
      'updateStage',
      'setTasksByTypeFilters',
    ]),
    onGroupSelect(group) {
      this.setSelectedGroup(group);
      this.fetchCycleAnalyticsData();
    },
    onProjectsSelect(projects) {
      this.setSelectedProjects(projects);
      this.fetchCycleAnalyticsData();
    },
    onStageSelect(stage) {
      this.hideCustomStageForm();
      this.setSelectedStage(stage);
      this.fetchStageData(this.selectedStage.slug);
    },
    onShowAddStageForm() {
      this.showCustomStageForm();
    },
    onShowEditStageForm(initData = {}) {
      this.showEditCustomStageForm(initData);
    },
    onCreateCustomStage(data) {
      this.createCustomStage(data);
    },
    onUpdateCustomStage(data) {
      this.updateStage(data);
    },
    onRemoveStage(id) {
      this.removeStage(id);
    },
    onDurationStageSelect(stages) {
      this.updateSelectedDurationChartStages(stages);
    },
  },
  multiProjectSelect: true,
  dateOptions: [7, 30, 90],
  groupsQueryParams: {
    min_access_level: featureAccessLevel.EVERYONE,
  },
  projectsQueryParams: {
    per_page: PROJECTS_PER_PAGE,
    with_shared: false,
    order_by: LAST_ACTIVITY_AT,
    include_subgroups: true,
  },
  durationChartTooltipDateFormat: dateFormats.defaultDate,
  maxDateRange: DATE_RANGE_LIMIT,
};
</script>
<template>
  <div class="js-cycle-analytics">
    <div class="page-title-holder d-flex align-items-center">
      <h3 class="page-title">{{ __('Value Stream Analytics') }}</h3>
    </div>
    <div class="mw-100">
      <div
        class="mt-3 py-2 px-3 d-flex bg-gray-light border-top border-bottom flex-column flex-md-row justify-content-between"
      >
        <groups-dropdown-filter
          v-if="!hideGroupDropDown"
          class="js-groups-dropdown-filter dropdown-select"
          :query-params="$options.groupsQueryParams"
          :default-group="selectedGroup"
          @selected="onGroupSelect"
        />
        <projects-dropdown-filter
          v-if="shouldDisplayFilters"
          :key="selectedGroup.id"
          class="js-projects-dropdown-filter ml-md-1 mt-1 mt-md-0 dropdown-select"
          :group-id="selectedGroup.id"
          :query-params="$options.projectsQueryParams"
          :multi-select="$options.multiProjectSelect"
          :default-projects="selectedProjects"
          @selected="onProjectsSelect"
        />
        <div
          v-if="shouldDisplayFilters"
          class="ml-0 ml-md-auto mt-2 mt-md-0 d-flex flex-column flex-md-row align-items-md-center justify-content-md-end"
        >
          <date-range
            :start-date="startDate"
            :end-date="endDate"
            :max-date-range="$options.maxDateRange"
            :include-selected-date="true"
            class="js-daterange-picker"
            @change="setDateRange"
          />
        </div>
      </div>
    </div>
    <gl-empty-state
      v-if="shouldRenderEmptyState"
      :title="__('Value Stream Analytics can help you determine your team’s velocity')"
      :description="
        __(
          'Start by choosing a group to see how your team is spending time. You can then drill down to the project level.',
        )
      "
      :svg-path="emptyStateSvgPath"
    />
    <div v-else class="cycle-analytics mt-0">
      <gl-empty-state
        v-if="hasNoAccessError"
        class="js-empty-state"
        :title="__('You don’t have access to Value Stream Analytics for this group')"
        :svg-path="noAccessSvgPath"
        :description="
          __(
            'Only \'Reporter\' roles and above on tiers Premium / Silver and above can see Value Stream Analytics.',
          )
        "
      />
      <div v-else-if="!errorCode">
        <div v-if="isLoading">
          <gl-loading-icon class="mt-4" size="md" />
        </div>
        <div v-else>
          <summary-table class="js-summary-table" :items="summary" />
          <stage-table
            v-if="selectedStage"
            class="js-stage-table"
            :current-stage="selectedStage"
            :stages="activeStages"
            :medians="medians"
            :is-loading="isLoadingStage"
            :is-empty-stage="isEmptyStage"
            :is-saving-custom-stage="isSavingCustomStage"
            :is-creating-custom-stage="isCreatingCustomStage"
            :is-editing-custom-stage="isEditingCustomStage"
            :current-stage-events="currentStageEvents"
            :custom-stage-form-events="customStageFormEvents"
            :custom-stage-form-errors="customStageFormErrors"
            :labels="labels"
            :no-data-svg-path="noDataSvgPath"
            :no-access-svg-path="noAccessSvgPath"
            :can-edit-stages="hasCustomizableCycleAnalytics"
            @clearCustomStageFormErrors="clearCustomStageFormErrors"
            @selectStage="onStageSelect"
            @editStage="onShowEditStageForm"
            @showAddStageForm="onShowAddStageForm"
            @hideStage="onUpdateCustomStage"
            @removeStage="onRemoveStage"
            @createStage="onCreateCustomStage"
            @updateStage="onUpdateCustomStage"
          />
        </div>
      </div>
      <template v-if="shouldDisplayDurationChart">
        <template v-if="isDurationChartLoaded">
          <div class="mt-3 d-flex">
            <h4 class="mt-0">{{ s__('CycleAnalytics|Days to completion') }}</h4>
            <stage-dropdown-filter
              v-if="stages.length"
              class="ml-auto"
              :stages="stages"
              @selected="onDurationStageSelect"
            />
          </div>
          <scatterplot
            v-if="durationChartPlottableData"
            :x-axis-title="s__('CycleAnalytics|Date')"
            :y-axis-title="s__('CycleAnalytics|Total days to completion')"
            :tooltip-date-format="$options.durationChartTooltipDateFormat"
            :scatter-data="durationChartPlottableData"
            :median-line-data="durationChartMedianData"
          />
          <div v-else ref="duration-chart-no-data" class="bs-callout bs-callout-info">
            {{ __('There is no data available. Please change your selection.') }}
          </div>
        </template>
        <gl-loading-icon v-else-if="!isLoading" size="md" class="my-4 py-4" />
      </template>
      <template v-if="shouldDisplayTasksByTypeChart">
        <div class="js-tasks-by-type-chart">
          <div v-if="isTasksByTypeChartLoaded">
            <tasks-by-type-chart
              :chart-data="tasksByTypeChartData"
              :filters="selectedTasksByTypeFilters"
              :labels="labels"
              @updateFilter="setTasksByTypeFilters"
            />
          </div>
          <gl-loading-icon v-else size="md" class="my-4 py-4" />
        </div>
      </template>
    </div>
  </div>
</template>
