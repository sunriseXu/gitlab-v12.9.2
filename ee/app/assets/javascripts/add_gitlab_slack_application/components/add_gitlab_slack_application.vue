<script>
/* eslint-disable @gitlab/vue-i18n/no-bare-strings */
import { __ } from '~/locale';
import Flash from '~/flash';
import { redirectTo } from '~/lib/utils/url_utility';

import GitlabSlackService from '../services/gitlab_slack_service';

export default {
  props: {
    projects: {
      type: Array,
      required: false,
      default: () => [],
    },

    isSignedIn: {
      type: Boolean,
      required: true,
    },

    gitlabForSlackGifPath: {
      type: String,
      required: true,
    },

    signInPath: {
      type: String,
      required: true,
    },

    slackLinkPath: {
      type: String,
      required: true,
    },

    gitlabLogoPath: {
      type: String,
      required: true,
    },

    slackLogoPath: {
      type: String,
      required: true,
    },

    docsPath: {
      type: String,
      required: true,
    },
  },

  data() {
    return {
      popupOpen: false,
      selectedProjectId: this.projects && this.projects.length ? this.projects[0].id : 0,
    };
  },

  computed: {
    doubleHeadedArrowSvg() {
      return gl.utils.spriteIcon('double-headed-arrow');
    },

    arrowRightSvg() {
      return gl.utils.spriteIcon('arrow-right');
    },

    hasProjects() {
      return this.projects.length > 0;
    },
  },

  methods: {
    togglePopup() {
      this.popupOpen = !this.popupOpen;
    },

    addToSlack() {
      GitlabSlackService.addToSlack(this.slackLinkPath, this.selectedProjectId)
        .then(response => redirectTo(response.data.add_to_slack_link))
        .catch(() => Flash(__('Unable to build Slack link.')));
    },
  },
};
</script>

<template>
  <div class="text-center">
    <h1>{{ __('GitLab for Slack') }}</h1>
    <p>{{ __('Track your GitLab projects with GitLab for Slack.') }}</p>

    <div v-once class="prepend-top-20 append-bottom-20">
      <img :src="gitlabLogoPath" class="gitlab-slack-logo" />
      <div
        class="gitlab-slack-double-headed-arrow inline prepend-left-20 append-right-20"
        v-html="doubleHeadedArrowSvg"
      ></div>
      <img :src="slackLogoPath" class="gitlab-slack-logo" />
    </div>

    <button
      type="button"
      class="btn btn-red mx-auto js-popup-button prepend-top-default"
      @click="togglePopup"
    >
      {{ __('Add GitLab to Slack') }}
    </button>

    <div
      v-if="popupOpen"
      class="popup gitlab-slack-popup mx-auto prepend-top-20 text-center js-popup"
    >
      <div v-if="isSignedIn && hasProjects" class="inline">
        <strong>{{ __('Select GitLab project to link with your Slack team') }}</strong>

        <select
          v-model="selectedProjectId"
          class="js-project-select form-control prepend-top-10 append-bottom-10"
        >
          <option v-for="project in projects" :key="project.id" :value="project.id">
            {{ project.name }}
          </option>
        </select>

        <button type="button" class="btn btn-red float-right js-add-button" @click="addToSlack">
          {{ __('Add to Slack') }}
        </button>
      </div>

      <span v-else-if="isSignedIn && !hasProjects" class="js-no-projects">{{
        __("You don't have any projects available.")
      }}</span>

      <span v-else>
        You have to
        <a v-once :href="signInPath" class="js-gitlab-slack-sign-in-link">{{ __('log in') }}</a>
      </span>
    </div>

    <div class="center prepend-top-20 append-bottom-10 append-right-5 prepend-left-5">
      <img v-once :src="gitlabForSlackGifPath" class="gitlab-slack-gif" />
    </div>

    <div v-once class="text-center">
      <h3>{{ __('How it works') }}</h3>

      <div class="well gitlab-slack-well mx-auto">
        <code class="code mx-auto append-bottom-10"
          >/gitlab &lt;project-alias&gt; issue show &lt;id&gt;</code
        >
        <span>
          <div class="gitlab-slack-right-arrow inline append-right-5" v-html="arrowRightSvg"></div>
          Shows the issue with id <strong>&lt;id&gt;</strong>
        </span>

        <div class="prepend-top-default">
          <a v-once :href="docsPath">{{ __('More Slack commands') }}</a>
        </div>
      </div>
    </div>
  </div>
</template>
