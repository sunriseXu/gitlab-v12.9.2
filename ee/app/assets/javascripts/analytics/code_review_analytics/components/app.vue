<script>
import { mapState, mapActions } from 'vuex';
import { GlBadge, GlLoadingIcon, GlEmptyState, GlPagination } from '@gitlab/ui';
import MergeRequestTable from './merge_request_table.vue';

export default {
  components: {
    GlBadge,
    GlLoadingIcon,
    GlPagination,
    GlEmptyState,
    MergeRequestTable,
  },
  props: {
    projectId: {
      type: Number,
      required: true,
    },
    newMergeRequestUrl: {
      type: String,
      required: true,
    },
    emptyStateSvgPath: {
      type: String,
      required: true,
    },
  },
  computed: {
    ...mapState({
      isLoading: 'isLoading',
      perPage: state => state.pageInfo.perPage,
      totalItems: state => state.pageInfo.total,
      page: state => state.pageInfo.page,
    }),
    currentPage: {
      get() {
        return this.page;
      },
      set(newVal) {
        this.setPage(newVal);
        this.fetchMergeRequests();
      },
    },
  },
  created() {
    this.setProjectId(this.projectId);
    this.fetchMergeRequests();
  },
  methods: {
    ...mapActions(['setProjectId', 'fetchMergeRequests', 'setPage']),
  },
};
</script>

<template>
  <div class="mt-2">
    <gl-loading-icon v-show="isLoading" size="md" class="mt-3" />
    <template v-if="!isLoading">
      <gl-empty-state
        v-if="!totalItems"
        :title="__(`You don't have any open merge requests`)"
        :primary-button-text="__('New merge request')"
        :primary-button-link="newMergeRequestUrl"
        :svg-path="emptyStateSvgPath"
      >
        <template #description>
          <div class="text-center">
            <p>
              {{
                __(
                  'Code Review Analytics displays a table of open merge requests considered to be in code review. There are currently no merge requests in review for this project and/or filters.',
                )
              }}
            </p>
          </div>
        </template>
      </gl-empty-state>
      <template v-else>
        <div>
          <span class="font-weight-bold">{{ __('Merge Requests in Review') }}</span>
          <gl-badge pill>{{ totalItems }}</gl-badge>
        </div>
        <merge-request-table />
        <gl-pagination
          v-model="currentPage"
          :per-page="perPage"
          :total-items="totalItems"
          align="center"
          class="w-100"
        />
      </template>
    </template>
  </div>
</template>
