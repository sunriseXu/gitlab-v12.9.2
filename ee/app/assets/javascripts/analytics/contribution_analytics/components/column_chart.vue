<script>
import { GlColumnChart } from '@gitlab/ui/dist/charts';
import { getSvgIconPathContent } from '~/lib/utils/icon_utils';
import ResizableChartContainer from '~/vue_shared/components/resizable_chart/resizable_chart_container.vue';

const CHART_HEIGHT = 220;

export default {
  components: {
    GlColumnChart,
    ResizableChartContainer,
  },
  props: {
    chartData: {
      type: Array,
      required: true,
    },
    xAxisTitle: {
      type: String,
      required: false,
      default: '',
    },
    yAxisTitle: {
      type: String,
      required: false,
      default: '',
    },
  },
  data() {
    return {
      svgs: {},
    };
  },
  computed: {
    dataZoomConfig() {
      const handleIcon = this.svgs['scroll-handle'];

      return handleIcon ? { handleIcon } : {};
    },
    chartOptions() {
      return {
        dataZoom: [this.dataZoomConfig],
      };
    },
    seriesData() {
      return { full: this.chartData };
    },
  },
  methods: {
    setSvg(name) {
      return getSvgIconPathContent(name)
        .then(path => {
          if (path) {
            this.$set(this.svgs, name, `path://${path}`);
          }
        })
        .catch(e => {
          // eslint-disable-next-line no-console, @gitlab/i18n/no-non-i18n-strings
          console.error('SVG could not be rendered correctly: ', e);
        });
    },
    onChartCreated() {
      this.setSvg('scroll-handle');
    },
  },
  height: CHART_HEIGHT,
};
</script>

<template>
  <resizable-chart-container>
    <gl-column-chart
      slot-scope="{ width }"
      v-bind="$attrs"
      :width="width"
      :height="$options.height"
      :data="seriesData"
      :x-axis-title="xAxisTitle"
      :y-axis-title="yAxisTitle"
      x-axis-type="category"
      :option="chartOptions"
      @created="onChartCreated"
    />
  </resizable-chart-container>
</template>
