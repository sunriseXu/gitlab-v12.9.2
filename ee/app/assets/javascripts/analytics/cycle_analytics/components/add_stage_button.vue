<script>
export default {
  props: {
    active: {
      type: Boolean,
      required: true,
    },
  },
  computed: {
    activeClass() {
      return 'active font-weight-bold border-style-solid border-color-blue-300';
    },
    inactiveClass() {
      return 'bg-transparent border-style-dashed border-color-default';
    },
  },
};
</script>
<template>
  <li
    :class="[active ? activeClass : inactiveClass]"
    class="js-add-stage-button stage-nav-item ml-2 mb-1 rounded d-flex justify-content-center border-width-1px"
    @click.prevent="$emit('showform')"
  >
    {{ s__('CustomCycleAnalytics|Add a stage') }}
  </li>
</template>
