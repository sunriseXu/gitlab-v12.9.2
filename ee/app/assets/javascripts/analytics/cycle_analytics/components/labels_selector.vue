<script>
import { GlDropdown, GlDropdownItem } from '@gitlab/ui';

export default {
  name: 'LabelsSelector',
  components: {
    GlDropdown,
    GlDropdownItem,
  },
  props: {
    labels: {
      type: Array,
      required: true,
    },
    selectedLabelId: {
      type: Number,
      required: false,
      default: null,
    },
  },
  computed: {
    selectedLabel() {
      const { selectedLabelId, labels } = this;
      if (!selectedLabelId || !labels.length) return null;
      return labels.find(({ id }) => id === selectedLabelId);
    },
  },
  methods: {
    labelTitle(label) {
      // there are 2 possible endpoints for group labels
      // one returns label.name the other label.title
      return label?.name || label.title;
    },
    isSelectedLabel(id) {
      return this.selectedLabelId && id === this.selectedLabelId;
    },
  },
};
</script>
<template>
  <gl-dropdown class="w-100" toggle-class="overflow-hidden">
    <template slot="button-content">
      <span v-if="selectedLabel">
        <span
          :style="{ backgroundColor: selectedLabel.color }"
          class="d-inline-block dropdown-label-box"
        >
        </span>
        {{ labelTitle(selectedLabel) }}
      </span>
      <span v-else>{{ __('Select a label') }}</span>
    </template>
    <gl-dropdown-item :active="!selectedLabelId" @click.prevent="$emit('clearLabel')"
      >{{ __('Select a label') }}
    </gl-dropdown-item>
    <gl-dropdown-item
      v-for="label in labels"
      :key="label.id"
      :active="isSelectedLabel(label.id)"
      @click.prevent="$emit('selectLabel', label.id)"
    >
      <span :style="{ backgroundColor: label.color }" class="d-inline-block dropdown-label-box">
      </span>
      {{ labelTitle(label) }}
    </gl-dropdown-item>
  </gl-dropdown>
</template>
