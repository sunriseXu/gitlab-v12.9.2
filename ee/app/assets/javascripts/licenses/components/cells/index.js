import Cell from './cell.vue';
import HeaderCell from './header_cell.vue';
import InfoCell from './info_cell.vue';
import DateCell from './date_cell.vue';
import SkeletonCell from './skeleton_cell.vue';
import SkeletonHeaderCell from './skeleton_header_cell.vue';

export { Cell, HeaderCell, InfoCell, DateCell, SkeletonCell, SkeletonHeaderCell };
