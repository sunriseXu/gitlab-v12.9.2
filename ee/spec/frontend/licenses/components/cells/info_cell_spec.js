import { shallowMount } from '@vue/test-utils';
import { InfoCell } from 'ee/licenses/components/cells';

describe('InfoCell', () => {
  let wrapper;
  const defaultProps = {
    title: 'title',
    value: 'value',
    popoverContent: 'popoverContent',
  };

  function createComponent(props, slots) {
    const propsData = Object.assign({}, defaultProps, props);

    wrapper = shallowMount(InfoCell, {
      propsData,
      slots,
    });
  }

  afterEach(() => {
    if (wrapper) wrapper.destroy();
  });

  it('renders a title and string value with an info popover through props', () => {
    createComponent();

    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders a number value', () => {
    createComponent({ value: 100 });

    expect(wrapper.element).toMatchSnapshot();
  });
});
