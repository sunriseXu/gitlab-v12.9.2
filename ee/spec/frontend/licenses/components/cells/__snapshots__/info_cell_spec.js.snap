// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCell renders a number value 1`] = `
<cell-stub
  class="license-info-cell"
  isflexible="true"
  value="100"
>
  <template>
    <span
      class="mr-2 text"
    >
      title
    </span>
     
    <button
      class="btn-link information-target"
      type="button"
    >
      <icon-stub
        class="icon d-block"
        name="information"
        size="16"
      />
    </button>
     
    <gl-popover-stub
      content="popoverContent"
      cssclasses=""
      placement="bottom"
      triggers="hover"
    />
  </template>
</cell-stub>
`;

exports[`InfoCell renders a title and string value with an info popover through props 1`] = `
<cell-stub
  class="license-info-cell"
  isflexible="true"
  value="value"
>
  <template>
    <span
      class="mr-2 text"
    >
      title
    </span>
     
    <button
      class="btn-link information-target"
      type="button"
    >
      <icon-stub
        class="icon d-block"
        name="information"
        size="16"
      />
    </button>
     
    <gl-popover-stub
      content="popoverContent"
      cssclasses=""
      placement="bottom"
      triggers="hover"
    />
  </template>
</cell-stub>
`;
